README.md

```diff
@@ -9,22 +9,24 @@
 
 [12-factor]: https://12factor.net/config
 
- - Set defaults
- - Set explicit values (to programmatically override)
- - Read from [JSON], [TOML], [YAML], [INI], [RON], [JSON5] files
- - Read from environment
- - Loosely typed — Configuration values may be read in any supported type, as long as there exists a reasonable conversion
- - Access nested fields using a formatted path — Uses a subset of JSONPath; currently supports the child ( `redis.port` ) and subscript operators ( `databases[0].name` )
-
-[JSON]: https://github.com/serde-rs/json
-[TOML]: https://github.com/toml-lang/toml
-[YAML]: https://github.com/chyh1990/yaml-rust
-[INI]: https://github.com/zonyitoo/rust-ini
-[RON]: https://github.com/ron-rs/ron
-[JSON5]: https://github.com/callum-oakley/json5-rs
-
-Please note that this library can not be used to write changed configuration
-values back to the configuration file(s)!
+-   Set defaults
+-   Set explicit values (to programmatically override)
+-   Read from [JSON], [TOML], [YAML], [INI], [RON], [JSON5] files
+-   Read from environment
+-   Loosely typed — Configuration values may be read in any supported type, as long as there exists a reasonable conversion
+-   Access nested fields using a formatted path — Uses a subset of JSONPath; currently supports the child ( `redis.port` ) and subscript operators ( `databases[0].name` )
+
+[json]: https://github.com/serde-rs/json
+[toml]: https://github.com/toml-lang/toml
+[yaml]: https://github.com/chyh1990/yaml-rust
+[ini]: https://github.com/zonyitoo/rust-ini
+[ron]: https://github.com/ron-rs/ron
+[json5]: https://github.com/callum-oakley/json5-rs
```

Comment:
Why was this changed?

Reply:
Unintentional change as I was trying to find the best place to add that the crate is case insensitive.

Reply:
No I think you changed some whitespace here. For example in c767d28 and in 0a2e8eb I can see whitespace change.

Sorry for being so persistent on this, I just want to be sure that the diff only touches what needs to be touched, because a larger diff makes harder to review. Right now it is not that much, but I hope you see the point: If a PR runs for weeks and the diff gets more and more overhead, this results in a discouraging amount of review work :laughing: 

If you don't want to continue because of my persistence, I can fully understand this and, if you're okay, I can take over this PR!

Reply:
> No I think you changed some whitespace here. For example in https://github.com/mehcode/config-rs/commit/c767d285d694f43539e4e98b6580024644cab22a and in https://github.com/mehcode/config-rs/commit/0a2e8ebcfe3951ea33ab99a3782f9193dd87d9d3 I can see whitespace change.

Yes I see what you mean. I think that has to do with the extension I am using when I hit save, my bad.

> Sorry for being so persistent on this, I just want to be sure that the diff only touches what needs to be touched, because a larger diff makes harder to review. Right now it is not that much, but I hope you see the point...

You are absolutely right, and at the time when I pushed this I just didn't notice these extra unnecessary changes in my commit. Thank you for pointing them out, I am gonna try and amend them later.